iree/compiler/Dialect/Flow/Transforms/ConvertToFlowTensorOps.cpp

```diff
@@ -30,11 +31,154 @@ namespace iree_compiler {
 namespace IREE {
 namespace Flow {
 
-namespace {
-/// Convert subtensor operation to flow.tensor.slice if
+/// An operation that uses `offsets`, `sizes` and `strides` (i.e. implements the
+/// `OffsetSizeAndStrideInterface` can be mapped to flow operations that
+/// eventually map to DMA operations if)
 /// - all offsets apart from the first one are 0
 /// - all the sizes apart from the first match the sizes of the source
 /// - all strides are 1.
+static bool isOffsetSizeAndStrideMappableToFlow(ArrayRef<OpFoldResult> offsets,
+                                                ArrayRef<OpFoldResult> sizes,
+                                                ArrayRef<OpFoldResult> strides,
+                                                ArrayRef<int64_t> baseShape) {
+  if (offsets.size() != baseShape.size()) {
+    // Unhanded rank-reducing case.
+    return false;
+  }
+  for (unsigned dim : llvm::seq<unsigned>(0, offsets.size())) {
```

Comment:
We should use `auto` here because `llvm::seq` spells the type.
```suggestion
  for (auto dim : llvm::seq<unsigned>(0, offsets.size())) {
```